crates/forge_domain/src/workflow.rs

```diff
@@ -128,6 +128,14 @@ pub struct Workflow {
     #[merge(strategy = crate::merge::option)]
     pub max_tool_failure_per_turn: Option<usize>,
 
+    /// Maximum execution time in seconds for a single tool call.
+    /// Controls how long a tool can run before being terminated.
+    /// Default is 300 seconds.
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[merge(strategy = crate::merge::option)]
+    pub tool_timeout_seconds: Option<u64>,
```

Comment:
I think it be better to implement this as an env feature instead of a workflow feature. Lets add this in `Environment`